src/main/java/org/testng/internal/Invoker.java

```diff
@@ -579,6 +580,29 @@ private ITestResult invokeMethod(
       ITestNGMethod[] afterMethods,
       ConfigurationGroupMethods groupMethods,
       FailureContext failureContext) {
+
+    if(System.getProperties().containsKey("testng.newInstancePerMethod") ||
+       System.getenv().containsKey("TESTNG_NEW_INSTANCE_PER_METHOD") ||
+       instance.getClass().isAnnotationPresent(NewInstancePerMethod.class)) {
+      boolean tryConstructor = true;
+      if (Cloneable.class.isAssignableFrom(instance.getClass())) {
+        try {
+          instance = instance.getClass().getDeclaredMethod("clone").invoke(instance);
+          tryConstructor = false;
+        } catch (NoSuchMethodException e) {
```

Comment:
trivial : can this be changed to `} catch (NoSuchMethodException | IllegalStateException | InvocationTargetException e) {` or perhaps just catch `Exception` (since we aren't doing anything with the exception apart from gobbling it and moving forward). On second thoughts perhaps we should throw a `TestNGException` to let the user know that the instantiation failed so that they can take corrective action, else they would basically reach out stating that they are using this feature, have turned on the feature but it still doesn't work :)